### PR TITLE
fix Amount class

### PR DESCRIPTION
**Pull request by kewde**
math.floor being crazy
fix net accounting rounding error

### CODE REVIEW

**Comment by kewde** [comment]
@allienworks

File: src/app/wallet/shared/transaction.model.ts [hunk 1]
```diff
@@ -67,18 +67,20 @@ export class Transaction implements Deserializable {
...
+  public getNetAmount(): number {
     const amount: number = +this.getAmountObject().getAmount();
 
     /* If fee undefined then just return amount */
     if (this.fee === undefined) {
       return amount;
     /* sent */
     } else if (amount < 0) {
-      return amount + this.fee;
+      const net = new Amount(+amount + (+this.fee)).getAmount();
```
Comment (rynomster) [inline src/app/wallet/shared/transaction.model.ts]: Can be one return, and next return looks the same?

File: src/app/shared/util/utils.ts [hunk 2]
```diff
@@ -16,7 +16,11 @@ export class Amount {
    * 25.9 -> '25'
    */
   public getIntegerPart(): number {
-    return Math.floor(this.amount);
+    /* Math floor is crazy.
+     * Math.floor(-0.2) -> -1.2, so fix it by adding 1 if negative &
+     * Math.floor(-9) -> -9, so only add 1 if no dot exists.
+     */
+    return Math.floor(this.amount) + Number(this.amount < 0 && this.ifDotExist());
```
Comment (rynomster) [inline src/app/shared/util/utils.ts]: Why not use `Math.trunc`?

Comment (kewde) [reply src/app/shared/util/utils.ts]: That's indeed a no-brainer, fixing.

File: src/app/wallet/shared/transaction.model.ts (hunk 1 above)
Comment (kewde) [reply src/app/wallet/shared/transaction.model.ts]: I'll smash the two lines into one, but I prefer to keep the else if structure, I'm not very fond of inline ( ? .. : ..) stuff. Makes readability and documenting it tough :/

**Comment by rynomster** [comment]
My bad, I didn't see the + and the minus.. Looked like 2 identical lines

**Pull request merged**